File: 프로그래머스/괄호 회전하기.java
import java.util.*;

class Solution {
    public int solution(String s) {
        int answer = 0;
        
        for(int i = 0; i < s.length(); i++){
            s = s.substring(1,s.length()) + s.substring(0,1);
            if(check(s))
                answer++;
        }
        return answer;
    }
    public static boolean check(String str){
        Stack<Character> stack = new Stack<>();
        
        for(int i = 0; i < str.length(); i++){
            char c = str.charAt(i);
            if(c == '(' || c == '[' || c == '{'){
                stack.push(c);
            }else{
                if(stack.isEmpty()) {
                    return false;
                }else{
                    char p = stack.pop();
                    if(c == ')' && p == '(')
                        continue;
                    else if(c == ']' && p == '[')
                        continue;
                    else if(c == '}' && p == '{')
                        continue;
                    else
                        return false;
                }
            }
        }
        
        if(!stack.isEmpty())
            return false;
        else
            return true;
    }
}

File: 프로그래머스/괄호 회전하기.py
def check(str):
    stack = []
    for i in range(0, len(str)):
        c = str[i]
        if c == '(' or c == '[' or c == '{' :
            stack.append(c)
        else:
            if len(stack) == 0:
                return False
            else:
                p = stack.pop()
                if c == ')' and p == '(':
                    continue
                elif c == ']' and p == '[':
                    continue
                elif c == '}' and p == '{':
                    continue
                else:
                    return False
    if len(stack) != 0:
        return False
    else:
        return True
    
def solution(s):
    answer = 0
    for i in range(0, len(s)):
        s = s[1:] + s[0];
        if check(s):
            answer += 1
    return answer

